scripts/repl.js

```diff
@@ -94,31 +132,29 @@
       'loose': $checkbox($loose)
     });
 
-    // Defaults
-    options.experimental = true;
-    options.playground = true;
-    options.loose = false;
-    options.evaluate = true;
-
-
+    // Merge in defaults
+    var defaults = {
+      experimental : true,
+      playground : true,
+      loose : false,
+      evaluate : true
+    };
+    
+    _.assign(options, defaults);
+                
     return options;
   }
 
   /*
    * 6to5 Web REPL
    */
-  function REPL () {
-    var state = UriUtils.parseQuery() || {};
-    
-    if (window.localStorage) {
-        try {
-          var storedState = localStorage.getItem('replState');
-          if (storedState) {
-            state = JSON.parse(storedState);  
-          }
-        } catch(e) {}
+  function REPL (storage) {
+    var state = {};
+    if (storage) {
+      _.assign(state, storage.get('replState'));
```

Comment:
I think it'd be better if the REPL just created its own storage. Something like:

```js
function REPL () {
  this.storage = new StorageService();
  var state = storage.get('replState') || {};
```